src/auxinfo/participant.rs

```diff
@@ -176,12 +178,18 @@ impl ProtocolParticipant for AuxInfoParticipant {
 }
 
 impl InnerProtocolParticipant for AuxInfoParticipant {
-    type Context = Vec<ParticipantIdentifier>;
+    type Context = ProtocolContext;
 
     fn retrieve_context(&self) -> <Self as InnerProtocolParticipant>::Context {
-        let mut ids = self.all_participants();
-        ids.sort();
-        ids
+        let mut participants = self.all_participants();
+        participants.sort();
+        let generator = CurvePoint::GENERATOR;
+        let order = k256_order();
+        ProtocolContext {
+            participants,
+            generator,
+            order,
+        }
```

Comment:
suggestion: this code gets repeated for every participant. Can you write a function on `ProtocolContext` that does this generically for an `InnerProtocolParticipant`? Like
```rust
fn protocol_context<P: InnerProtocolParticipant>(p: &P) -> Self {
    // this code here
}